File: 4.OOPinJava/4.2.Encapsulation/Team.java
// Task 2:
// Create a class `Team` with an inner class `Player`.
// `Team` should manage a list of `Player` objects with methods to add, remove, and list players.

import java.util.ArrayList;
import java.util.List;

class Team {
    private List<Player> roster = new ArrayList<>();

    // TODO: implement addPlayer(String name, int number)
    // TODO: implement removePlayer(String name)
    // TODO: implement displayPlayers()

    public class Player {
        private String name;
        private int number;
        public Player(String name, int number) {
            this.name = name;
            this.number = number;
        }
        // getters...
    }
}

public class Main {
    public static void main(String[] args) {
        Team team = new Team();
        team.addPlayer("Alice", 10);
        team.addPlayer("Bob", 23);

        System.out.println("Roster:");
        team.displayPlayers();
        // Expected:
        // Name: Alice, Number: 10
        // Name: Bob, Number: 23

        team.removePlayer("Alice");
        System.out.println("\nAfter removal:");
        team.displayPlayers();
        // Expected:
        // Name: Bob, Number: 23
    }
}

File: 3.OOPinPython/3.2.Encapsulation/team.py
"""
Task 2:
Create a class `Team` with an inner class `Player`.
`Team` should manage a list of `Player` objects with methods to add, remove, and list players.
Implement private class Player field `passport`.
"""

from typing import List

class Team:
    # TODO: add roster field and constructor

    def add_player(self, name: str, number: int) -> None:
        # TODO: implement add_player
        pass

    def remove_player(self, name: str) -> None:
        # TODO: implement remove_player
        pass

    def display_players(self) -> None:
        # TODO: implement display_players
        pass

    class Player:
        # TODO: implement Player class with passport class fieldname and number instance fields
        pass


def main():
    team = Team()
    team.add_player("Alice", 10)
    team.add_player("Bob", 23)

    print("Roster:")
    team.display_players()
    # Expected:
    # Name: Alice, Number: 10
    # Name: Bob, Number: 23

    team.remove_player("Alice")
    print("\nAfter removal:")
    team.display_players()
    # Expected:
    # Name: Bob, Number: 23

if __name__ == "__main__":
    main() 